ingest/usecase/plugins/orderbook/claimbot/order.go

```diff
@@ -0,0 +1,183 @@
+package claimbot
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/osmosis-labs/osmosis/osmomath"
+	"github.com/osmosis-labs/sqs/domain"
+	"github.com/osmosis-labs/sqs/domain/mvc"
+	orderbookdomain "github.com/osmosis-labs/sqs/domain/orderbook"
+	orderbookgrpcclientdomain "github.com/osmosis-labs/sqs/domain/orderbook/grpcclient"
+	"github.com/osmosis-labs/sqs/log"
+
+	"go.uber.org/zap"
+)
+
+type order struct {
+	Orderbook domain.CanonicalOrderBooksResult
+	Orders    orderbookdomain.Orders
+	Err       error
+}
+
+// processOrderbooksAndGetClaimableOrders processes a list of orderbooks and returns claimable orders for each.
+func processOrderbooksAndGetClaimableOrders(
+	ctx context.Context,
+	fillThreshold osmomath.Dec,
+	orderbooks []domain.CanonicalOrderBooksResult,
+	orderbookRepository orderbookdomain.OrderBookRepository,
+	orderBookClient orderbookgrpcclientdomain.OrderBookClient,
+	orderbookusecase mvc.OrderBookUsecase,
```

Comment:
Related to my other comment, we should only need to pass orderbook use case and proxy requests to the repository through it

Reply:
Thanks, addressed as per your suggestion!